apitools/org.eclipse.pde.api.tools/src/org/eclipse/pde/api/tools/internal/APIFileGenerator.java

```diff
@@ -212,39 +212,27 @@ public void generateAPIFile() {
 					File currentManifest = new File(allManifestFile);
 					Set<String> currentApiPackages = null;
 					if (currentManifest.exists()) {
-						BufferedInputStream inputStream = null;
-						ZipFile zipFile = null;
 						try {
 							if (isZipJarFile(currentManifest.getName())) {
-								zipFile = new ZipFile(currentManifest);
-								final ZipEntry entry = zipFile.getEntry("META-INF/MANIFEST.MF"); //$NON-NLS-1$
-								if (entry != null) {
-									inputStream = new BufferedInputStream(zipFile.getInputStream(entry));
+								try (ZipFile zipFile = new ZipFile(currentManifest)) {
+									final ZipEntry entry = zipFile.getEntry("META-INF/MANIFEST.MF"); //$NON-NLS-1$
+									if (entry != null) {
+										try (BufferedInputStream inputStream = new BufferedInputStream(
+												zipFile.getInputStream(entry))) {
```

Comment:
`ManifestElement.parseBundleManifest` already wraps the stream passed into a `BufferedInputStream` so the wrapping can be removed here and below.

And in general Inputstreams returned by a ZipFile are closed when the ZipFile is closed, so you can skip wrapping them in a try-with-resources here and below.

Reply:
done